anadi/__main__.py

```diff
@@ -7,24 +7,29 @@
 from anadi.models.confs import SettingsApp, SettingsDB
 
 parser = argparse.ArgumentParser(prog="AnadiCSV", description="Handle CVS file using SQL language")
-parser.add_argument("-d", "--dir", type=str, help="CSV file dir", required=True)
+parser.add_argument("-d", "--dir", type=str, help="CSV file dir", required=False)
 parser.add_argument("-f", "--file", type=str, help="CSV file name", required=False)
 parser.add_argument("-q", "--query", type=str, help="SQL query to execute", required=False)
 
 
 def run():
     args = parser.parse_args()
-
+    
+    if not args.dir and not(args.file and args.query) or args.dir and (args.file and args.query):
+        raise ValueError("Error: Either --dir must be provided, or both --file and --query must be provided.")
+    if args.dir and args.file:
+        raise ValueError("Error: Cannot user both --dir  and --file together")        
     try:
         app = AnadiApp()
-        app.init(args.dir, os.path.abspath(os.path.expanduser(ANADI_CONF_FILE)))
-        if args.query:
+        if args.dir:
+            app.init(args.dir, os.path.abspath(os.path.expanduser(ANADI_CONF_FILE)))
+            app.run()
+        elif args.file and args.query:
             csvdb = CSVDB()
             csvdb.load(args.file,SettingsDB())
             result = csvdb.raw_sql(args.query)
             print(result)
-        else:
-            app.run()
+        
```

Comment:
if *args.dir* and *args.query* are set but *args.file* not we execute *args.dir* anyway and not error should be shown.

So or the condition is improved (but if we add another option that check became bigger) or we decide to remove this check and error and let the -d option win other the others.

Any idea ?

Reply:
Initially, I let the args.dir win. If args.dir  and args.file are set not args.query I opened the editor. But the user experience doesn't seem good.
Its either directory in the editor or the file name with the query.
We could make this clear in the bash script help flag for better usability. 

Works??